great_expectations/dataset/pandas_dataset.py

```diff
@@ -301,27 +301,31 @@ def expect_column_values_to_not_be_null(self, column,
         series = self[column]
         boolean_mapped_null_values = series.isnull()
 
-        element_count = int(len(series))
-        nonnull_values = series[boolean_mapped_null_values==False]
-        nonnull_count = int((boolean_mapped_null_values==False).sum())
+        element_count = len(series)
+        success_count = sum(~boolean_mapped_null_values)
+        unexpected_count = element_count - success_count
 
-        boolean_mapped_success_values = boolean_mapped_null_values==False
-        success_count = boolean_mapped_success_values.sum()
-
-        unexpected_list = [None for i in list(series[(boolean_mapped_success_values==False)])]
-        unexpected_index_list = list(series[(boolean_mapped_success_values==False)].index)
-        unexpected_count = len(unexpected_list)
+        unexpected_list = [None for i in range(unexpected_count)]
+        unexpected_index_list = boolean_mapped_null_values[boolean_mapped_null_values].index
```

Comment:
`series[boolean_mapped_null_values].index` for clarity? (please confirm that's also right).

Reply:
You're totally right, it produces the same result but using series[].index is clearer in it's purpose.